lms/djangoapps/grades/tests/test_models.py

```diff
@@ -0,0 +1,180 @@
+"""
+Unit tests for grades models.
+"""
+from base64 import b64encode
+from collections import OrderedDict
+import ddt
+from hashlib import sha256
+import json
+from mock import patch
+
+from django.core.exceptions import ValidationError
+from django.test import TestCase
+from opaque_keys.edx.locator import CourseLocator, BlockUsageLocator
+
+from lms.djangoapps.grades.models import BlockRecord, VisibleBlocks, PersistentSubsectionGrade
+
+
+class GradesModelTestCase(TestCase):
+    """
+    Base class for common setup of grades model tests.
+    """
+    def setUp(self):
+        super(GradesModelTestCase, self).setUp()
+        self.course_key = CourseLocator(
+            org='some_org',
+            course='some_course',
+            run='some_run'
+        )
+        self.locator_a = BlockUsageLocator(
+            course_key=self.course_key,
+            block_type='problem',
+            block_id='block_id_a'
+        )
+        self.locator_b = BlockUsageLocator(
+            course_key=self.course_key,
+            block_type='problem',
+            block_id='block_id_b'
+        )
+        self.record_a = BlockRecord(unicode(self.locator_a), 1, 10)
+        self.record_b = BlockRecord(unicode(self.locator_b), 1, 10)
+
+
+@ddt.ddt
+class BlockRecordTest(GradesModelTestCase):
+    """
+    Test the BlockRecord model.
+    """
+    def setUp(self):
+        super(BlockRecordTest, self).setUp()
+
+    def test_creation(self):
+        """
+        Tests creation of a BlockRecord.
+        """
+        weight = 1
+        max_score = 10
+        record = BlockRecord(
+            self.locator_a,
+            weight,
+            max_score,
+        )
+        self.assertEqual(record.locator, self.locator_a)
+
+    @ddt.data(
+        (0, 0, "0123456789abcdef"),
+        (1, 10, 'totally_a_real_block_key'),
+        ("BlockRecord is", "a dumb data store", "with no validation"),
+    )
+    @ddt.unpack
+    def test_serialization(self, weight, max_score, block_key):
+        """
+        Tests serialization of a BlockRecord using the to_dict() method.
+        """
+        record = BlockRecord(block_key, weight, max_score)
+        expected = OrderedDict([
+            ("locator", block_key),
+            ("weight", weight),
+            ("max_score", max_score),
+        ])
+        self.assertEqual(expected, record._asdict())
+
+
+class VisibleBlocksTest(GradesModelTestCase):
+    """
+    Test the VisibleBlocks model.
+    """
+    def setUp(self):
+        super(VisibleBlocksTest, self).setUp()
+
+    def test_creation(self):
+        """
+        Happy path test to ensure basic create functionality works as expected.
```

Comment:
Is a "sad path" test needed here to ensure expected behavior for any edge cases?